AWS Build Cloud Account Id Artefacts.yml

```diff
@@ -1,288 +1,295 @@
-        TestImageName: $(BuildImage.amiID)
-        TestCloudName: AWS
-        AtomicBuild: $(atomic-build)
-        TestVmName: Test$(System.JobDisplayName)
-        SG: $(System.JobDisplayName)
-        keypair: $(KeypairName)
-        keypairpath: $(keypair.secureFilePath)
-        VersionText: $(System.JobDisplayName)
-    inputs:
-      awsCredentials: $(awsCredentials)
-      regionName: $(regionName)
-      filePath: scripts/invoke-pester-tests.ps1
-      inlineScript: >2-
-
-        # Invoke Pester Tests
-
-        cd $(System.DefaultWorkingDirectory)\Tests
-
-        $result = Invoke-Pester -Script '.\Image*' -OutputFormat  NUnitXml -OutputFile '$(System.DefaultWorkingDirectory)\Test-Vm.xml' -PassThru
-
-        $result | Out-Default | Write-Host
-
-        if ($result.Result -eq "Failed") {
-            throw "Failed Tests Count: $($result.FailedCount)"
-        } else {
-            Write-Host "Tested the image successfully."
-        }
-  - task: PublishTestResults@2
-    displayName: Publish Test Results **/Test-*.xml
-    inputs:
-      testRunner: NUnit
-      testResultsFiles: '**/Test-*.xml'
-      testRunTitle: $(System.JobDisplayName)
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: Image URL'
-    inputs:
-      PathtoPublish: $(Build.ArtifactStagingDirectory)/$(System.JobDisplayName).txt
-      ArtifactName: aws
-      FileCopyOptions: ''
-  - task: AmazonWebServices.aws-vsts-tools.AWSPowerShellModuleScript.AWSPowerShellModuleScript@1
-    displayName: Deleting Security group and Instance
-    inputs:
-      awsCredentials: $(awsCredentials)
-      regionName: $(regionName)
-      arguments: -version "$(System.JobDisplayName)"
-      filePath: scripts/terminate-instance.ps1
-      inlineScript: >+
-        # Terminating the Instance
-
-        if('$(TerminateInstance)' -eq 'True') {
-           Write-Host "Removing the instance $(BuildImage.InstanceID) "
-            Remove-EC2Instance -InstanceId $(BuildImage.InstanceID) -Force
-        }
-
-        #Removing Vm
-
-        Write-Host "Removing the Vm"
-
-        Remove-EC2Instance -InstanceId  $(Vmtest.instanceID) -Force
-
-
-        # Deleting the Security Group
-
-        Write-Host "Deleting the security group"
-
-        Start-Sleep -Seconds 180
-         Remove-EC2SecurityGroup -GroupName 'w19d-15-0-$(VersionText-w19d-15-0)' -Force
-
-    
-
-...
+name: Build Cloud Accounts Artefacts 3.0-$(Build.BuildId)
+
+variables:
+  - group: "External IP Addresses"
+  - group: "Logins"
+  - name: regionName
+    value: ap-southeast-2
+  - name: atomic-build-ps
+    value: $false
+  - name: regionName
+    value: ap-southeast-2
+  - name: GitBranchPR
+    value: feature/aws-Build-Cloud-Account-Id-Artefacts-3.0
+  - name: GitUserName
+    value: priyadishah
+  - name: Windows-2019-AMI
+    value: LANSA Scalable License  w19d-15*
+
+resources:
+  repositories:
+  - repository: self
+    type: git
+    ref: feature/aws-Build-Cloud-Account-Id-Artefacts-3.0
```

Comment:
Strange to have a git reference in the YAML. Is there not a better place for this?

Reply:
I will keep it with GitBranchPR and GitUserName in pipeline variables.